sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,506 @@
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `N`.
+
+==== Fixed-size Array Overload
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Store API
+template<typename Group, typename OutputIteratorT, typename InputT,
+        std::size_t ElementsPerWorkItem, typename Properties = empty_properties_t>
+void group_store(Group g, sycl::span<InputT, ItemsPerWorkItem> in,
+                 OutputIteratorT out_iter,  Properties = {});
+
+}
+----
+
+._Preconditions_:
+* `out_iter` must be the same for all work-items in the group.
+* `in` must be a `sycl::span` made from a pointer to the private memory space.
+
+._Constraints_:
+* Only available if `Group` is a
+work-group or sub-group.
+* `OutputIteratorT` suppose to be a random access iterator.
+* `InputT` must be convertible to value type of `OutputIteratorT`.
+
+_Effects_: Stores `ElementsPerWorkItem` elements from `in` span to `out_iter`
+using the `g` group object.
+
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out_iter[g.get_local_linear_id() * ItemsPerWorkItem + i]` = `in[i];`
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `ItemsPerWorkItem`.
+
+
+=== Data Placement
+
+anchor:data_placement[]
+
+To specify a correct data placement for placing of resulting data
+there is a enum
+(proposed in link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]):
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Properties:
+enum class group_algorithm_data_placement {
+  blocked,
+  striped
+};
+
+}
+----
+
+Data placement layout example on group_load:
+
+* ElementsPerWorkItem = 4
+* 3 work-items in the group
+* input is: in_iter[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::blocked`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::striped`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+This extention adds a property that satisfies
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[SYCL Properties Extension]
+requirements to identify data_placement similar to
+link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]
+`input_data_placement` and `output_data_placement`:
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::property
+{
+    template<group_algorithm_data_placement type>
+    struct data_placement;
+}
+----
```

Comment:
This is not the right way to define a compile-time property.  It should be like this instead:

```
namespace sycl::ext::oneapi::experimental {

struct data_placement_key {
  template <group_algorithm_data_placement Type>
  using value_t =
      property_value<data_placement_key,
                     std::integral_constant<group_algorithm_data_placement, Type>>;
};

template <group_algorithm_data_placement Type>
inline constexpr data_placement_key::value_t<Type> data_placement;

inline constexpr data_placement_key::value_t<group_algorithm_data_placement::blocked> data_placement_blocked;
inline constexpr data_placement_key::value_t<group_algorithm_data_placement::striped> data_placement_striped;

} // namespace sycl::ext::oneapi::experimental
```

An example usage would then be like:

```
group_load(g, input, output_span, data_placement_blocked);
```

BTW, I agree with the previous comment that the name `group_algorithm_data_placement` is too long.  How about `data_placement_enum`.  We have a convention with other properties where we use the `_enum` suffix.

Reply:
Applied suggestion, I assume it is also need to be in experimental::property namespace?
And another thought: should data_placement also be a namespace then or is it too much?

The previous version was based on group_sort proposal, I assume it would also require similar update

Reply:
ok, got it, removed property:: namespace, thank you!